src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,711 @@
+      Bind(EVT_MOVABLE_CONTROL_DRAG_FINISHED, [this, line](const MovableControlEvent& event)
+      {
+         auto& effectList = RealtimeEffectList::Get(*mTrack);
+         if(event.GetSourceIndex() != event.GetTargetIndex())
+         {
+            assert(event.GetSourceIndex() >= 0 && event.GetSourceIndex() < effectList.GetStatesCount());
+            assert(event.GetTargetIndex() >= 0 && event.GetTargetIndex() <= effectList.GetStatesCount());
+            effectList.Reorder(event.GetSourceIndex(), event.GetTargetIndex());
+         }
+         line->Hide();
+      });
+   }
+
+   void SetTrack(const std::shared_ptr<Track>& track)
+   {
+      if(mTrack == track)
+         return;
+
+      wxWindowUpdateLocker freeze(this);
+
+      mTrack = track;
+      mEffectListContainer->DestroyChildren();
+
+      if(track)
+      {
+         const auto& effects = RealtimeEffectList::Get(*track);
+         for(int i = 0, count = effects.GetStatesCount(); i < count; ++i)
+            AddEffectRow(i);
+         
+         SetScrollRate(0, 20);
+      }
+      mAddEffect->Enable(!!track);
+      Layout();
+   }
+
+   void OnAddEffectClicked(const wxCommandEvent& event)
+   {
+      auto effectID = ShowSelectEffectMenu(dynamic_cast<wxWindow*>(event.GetEventObject()));
+      if(!effectID.empty())
+      {
+         auto& effects = RealtimeEffectList::Get(*mTrack);
+         if(auto effectState = effects.AddState(effectID))
+         {
+            assert(effects.GetStatesCount() > 0);//If effect added, then there should be at least one effect on the stack
+            AddEffectRow(static_cast<int>(effects.GetStatesCount()) - 1);
+            Layout();
+         }
+      }
+   }
+
+   void AddEffectRow(int effectStateIndex)
+   {
+      auto row = safenew ThemedWindowWrapper<RealtimeEffectControl>(mEffectListContainer, wxID_ANY);
+      row->SetBackgroundColorIndex(clrEffectListItemBackground);
+      row->SetEffect(mTrack, effectStateIndex);
+      row->SetMinSize(wxSize { 20, 50 });
+      mEffectListContainer->GetSizer()->Add(row, 0, wxEXPAND);
+   }
+
+};
+
+
+RealtimeEffectPanel::RealtimeEffectPanel(wxWindow* parent, wxWindowID id, const wxPoint& pos, const wxSize& size,
+   long style, const wxString& name)
+      : wxWindow(parent, id, pos, size, style, name)
+{
+   auto vSizer = std::make_unique<wxBoxSizer>(wxVERTICAL);
+   
+   auto header = safenew ThemedWindowWrapper<wxWindow>(this, wxID_ANY);
+   header->SetBackgroundColorIndex(clrMedium);
+   {
+      auto hSizer = std::make_unique<wxBoxSizer>(wxHORIZONTAL);
+      auto toggleEffects = safenew ThemedButtonWrapper<wxToggleButton>(header, wxID_ANY, wxEmptyString, wxDefaultPosition, wxDefaultSize, wxNO_BORDER);
+      toggleEffects->SetBitmapLabelIndex(bmpEffectOff);
+      toggleEffects->SetBitmapPressedIndex(bmpEffectOn);
+      toggleEffects->SetBackgroundColorIndex(clrMedium);
+      toggleEffects->SetMinSize(toggleEffects->GetBitmapLabel().GetSize());
+      mToggleEffects = toggleEffects;
```

Comment:
I see no callback bound to this toggle button.  Maybe bind an empty lambda with one more explicit TODO comment, like other TODOs in this commit.